### DISCUSSION TITLE
Thrown exception in TypeConverter not honored

### DISCUSSION THREAD

**Discussion by gilliduck** [spectreconsole/spectre.console#1296] [General]
**Information**
 - OS: all
 - Version: 0.47.0
 - Terminal: Windows Terminal

**Describe the bug**
The exception thrown inside of a custom `TypeConverter` is not honored. See below code for full example, but I'm throwing a `NotSupportedException` with a specific message. The message that is rendered in the console is not my exception message but instead a more generic message.

**To Reproduce**
Throw an exception with a custom message in a `TypeConverter` implementation used on a `CommandOption` or `CommandArgument`. Ensure that the exception is hit, note that the output message in the console is not the message from the exception (nor is stacktrace shown, i.e. the exception is eaten).

**Expected behavior**
Exception behavior is honored and the thrown exception is bubbled up appropriately. Message is show at the very least, ideally full exception behavior is honored.

**Additional context**
```
[CommandOption("-p|--platform")]
[Description("Filter by Platform(s)")]
[TypeConverter(typeof(PlatformConverter))]
[DefaultValue(null)]
public Platform[] Platforms { get; }
```

```
internal sealed class PlatformConverter : TypeConverter
{
	public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
	{
		if (value is string stringValue)
		{
			if (Enum.TryParse<Platform>(stringValue, true, out var asEnum))
			{
				return asEnum;
			}
		}
		
		throw new NotSupportedException($"Can't convert value to {nameof(TEnum)}. Valid options are {string.Join(", ", Enum.GetValues<Platform>())}");
	}
}
```

Output in console:
`Error: Failed to convert 'foo' to SuggestedActionType[].`

**Comment by nils-a** [comment]
@gilliduck The Exception is "honored". If you visualize the exception, you'll be able to see "your" exception as the inner exception.

My guess is what you really "wanted" is to display the message you set in the exception to the user. This is intentionally not done, as exceptions normally contain very technical descriptions, which most users of the application simply do not care about.

Is there a reason you are trying to use a custom exception containing valid values instead of implementing the parts of `TypeConverter` that were intended to do this? (Meaning [`TypeConverter.GetStandardValuesSupported`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.typeconverter.getstandardvaluessupported?view=net-7.0) and [`TypeConverter.GetStandardValues`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.typeconverter.getstandardvalues?view=net-7.0)

**Comment by VictorioBerra** [comment]
@gilliduck TEnum does not exist in that code snippet.

**Reply by gilliduck** [reply]
I assume at the time I wrote that it was a oversight during the translation from my original code to a MRE. However, it's been almost a year and I haven't worked on the project that involved that code snippet in almost as much time. I'd wager I meant for it to be `Platform` instead of `TEnum`.
